src/core/server/licensing/licensing_service_setup.ts

```diff
@@ -0,0 +1,191 @@
+    return this.license.mode;
+  }
+
+  get isBasic() {
+    return this.isActive && this.mode === 'basic';
+  }
+
+  get isNotBasic() {
+    return this.isActive && this.mode !== 'basic';
+  }
+
+  get reasonUnavailable() {
+    if (!this._license) {
+      return `[${this.clusterSource}] Elasticsearch cluster did not respond with license information.`;
+    }
+
+    if (this.error instanceof Error && (this.error as any).status === 400) {
+      return `X-Pack plugin is not installed on the [${this.clusterSource}] Elasticsearch cluster.`;
+    }
+
+    return this.error;
+  }
+
+  get signature() {
+    if (this._signature !== undefined) {
+      return this._signature;
+    }
+
+    this._signature = createHash('md5')
+      .update(JSON.stringify(this.toObject()))
+      .digest('hex');
+
+    return this._signature;
+  }
+
+  isOneOf(candidateLicenses: string | string[]) {
+    if (!Array.isArray(candidateLicenses)) {
+      candidateLicenses = [candidateLicenses];
+    }
+
+    return candidateLicenses.includes(this.license.mode);
+  }
+
+  meetsMinimumOf(minimum: LICENSE_TYPE) {
+    return minimum >= LICENSE_TYPE[this.mode as LicenseType];
+  }
+
+  check(pluginName: string, minimumLicenseRequired: LICENSE_TYPE | string) {
+    const minimum = toLicenseType(minimumLicenseRequired);
+
+    if (!this._license || !this.isAvailable) {
+      return {
+        status: LICENSE_STATUS.Unavailable,
+        message: i18n.translate('xpack.server.checkLicense.errorUnavailableMessage', {
+          defaultMessage:
+            'You cannot use {pluginName} because license information is not available at this time.',
+          values: { pluginName },
+        }),
+      };
+    }
+
+    const { type: licenseType } = this.license;
+
+    if (!this.meetsMinimumOf(minimum)) {
+      return {
+        status: LICENSE_STATUS.Invalid,
+        message: i18n.translate('xpack.server.checkLicense.errorUnsupportedMessage', {
+          defaultMessage:
+            'Your {licenseType} license does not support {pluginName}. Please upgrade your license.',
+          values: { licenseType, pluginName },
+        }),
+      };
+    }
+
+    if (!this.license.isActive) {
+      return {
+        status: LICENSE_STATUS.Expired,
+        message: i18n.translate('xpack.server.checkLicense.errorExpiredMessage', {
+          defaultMessage:
+            'You cannot use {pluginName} because your {licenseType} license has expired.',
+          values: { licenseType, pluginName },
+        }),
+      };
+    }
+
+    return { status: LICENSE_STATUS.Valid };
+  }
+
+  toObject() {
+    if (this.objectified) {
+      return this.objectified;
+    }
+
+    this.objectified = {
+      license: {
+        type: this.type,
+        isActive: this.isActive,
+        expiryDateInMillis: this.expiryDateInMillis,
+      },
+      features: [...this.featuresMap].map(([, feature]) => feature.toObject()),
+    };
+
+    return this.objectified;
+  }
+
+  feature(name: string) {
```

Comment:
Minor nit, but I'd find this method easier to understand if there was a verb in the name, e.g. `getFeature`.